### PR TITLE
Refactor/bank spendable

### PR DESCRIPTION
**Pull request by ccamel**
Similar to #1057. Refactor `bank_spendable_balances/2` predicate to use VFS-based implementation through `/bank/{address}/spendable/@` endpoint.

This is a breaking change : the `bank_spendable_balances/2` predicate now requires the address parameter to be instantiated. Previously, passing a variable would iterate over all accounts; now it raises an `instantiation_error`.

## Release Notes

* **New Features**
  * Added bank_spendable_balances/2 predicate enabling queries on spendable account balances; users can now retrieve the amount of coins available for immediate use in accounts.

* **Tests**
  * Added comprehensive test scenarios for the new spendable balances feature, covering valid queries, invalid addresses, and edge cases.

### CODE REVIEW

File: x/logic/fs/bank/fs.go [hunk 1]
```diff
@@ -48,17 +50,17 @@ func (f *vfs) Open(name string) (fs.File, error) {
...
+	addr, fetcher, err := f.validatePath(subpath)
 	if err != nil {
 		return nil, &fs.PathError{Op: "open", Path: name, Err: err}
 	}
 
 	bankKeeper, err := prolog.ContextValue[types.BankKeeper](f.ctx, types.BankKeeperContextKey, nil)
 	if err != nil {
-		return nil, &fs.PathError{Op: "open", Path: name, Err: fmt.Errorf("bank keeper not found in context: %w", err)}
+		return nil, &fs.PathError{Op: "open", Path: name, Err: fs.ErrNotExist}
 	}
```
[the remark this answers is not in the document]
Comment (ccamel) [reply x/logic/fs/bank/fs.go:64]: Mmm. Good catch. 🧐 This is indeed a subtle case.

I initially mapped it to `fs.ErrNotExist` to avoid leaking internal wiring details to the Prolog layer and query results. However, that is misleading, as this is not a "_file does not exist_" situation but an _environment/wiring_ failure.

I'll switch to returning a canonical `*fs.PathError` with a stable sentinel error (`vfs_unavailable`). This keeps the io/fs contract intact, avoids exposing internal details, and preserves a consistent error category on the Prolog side.

File: x/logic/tests/predicate/features_test.go [hunk 2]
```diff
@@ -405,3 +385,46 @@ func parseDocStringYaml(docString *godog.DocString, v any) error {
...
+		if len(row.Cells) != 2 {
+			return nil, nil, fmt.Errorf("each balance row must have exactly 2 cells (denom, amount)")
+		}
+
+		denom := row.Cells[0].Value
+		amountStr := row.Cells[1].Value
+
+		if denom == "" || amountStr == "" {
+			continue // Skip empty rows
+		}
```
[the remark this answers is not in the document]
Comment (ccamel) [reply x/logic/tests/predicate/features_test.go:411]: yup. Makes sense.
I'll fix it.

**Comment by ccamel** [comment]
@coderabbitai resolve
